index.html

```diff
@@ -280,30 +302,21 @@ <h1 class="collapse">Discounted Dining Finder: Places to Eat Out To Help Out</h1
 
         const x = Math.round((+result.lat - 49.0) / (12.0 / 160.0));
         const y = Math.round((+result.lon + 9) / (11.0 / 80.0));
-
+        let current = {"x": x, "y": y};
+	if (httpCache.location && (JSON.stringify(httpCache.location) == JSON.stringify(current))) {
+		renderTable(httpCache.pubs, result);
+                return ;
+	}
         d3.csv("pubgrid/" + x + "/" + x + "-" + y + ".csv")
             .then(function(pubs) {
-                showResults(pubs
-                    .map(function(a) {
-                        return {
-                            name: a.name,
-                            postcode: a.postcode,
-                            lat: a.lat,
-                            lon: a.lon,
-                            distance: distance(result, a)
-                        };
-                    })
-                    .filter(function(a) {
-                        return a.distance < (5 * 1609.34);
-                    })
-                    .sort(function(a, b) {
-                        return a.distance - b.distance;
-                    })
-                    .slice(0, 250));
+                    renderTable(pubs, result);
+                    renderMap(pubs);
```

Comment:
CLARIFICATION: In central London, a single tile has over 10,000 restaurants (see `pubgrid/33/33-64.csv`) - does that render ok?  I'm thinking older laptops or phones here.  Removing the `slice` also means that the table would be re-rendered with potentially a very large number of restaurants, which I'm not sure is such a great UX.

Reply:
This was coded on a 7 year old laptop and worked fine. Also tested on Moto g 5 (budget phone 3 years old) also fine. 

The table still had the slice 250 on it. It's just removed from the market render

Reply:
250 was removed by accident, added back to just the table view. The map does seem to handle it well